### ISSUE TITLE
Pylance isinstance bug ?

### ISSUE THREAD

**Issue by PASUNX** [microsoft/pylance-release#1499]
<h3> Environment data </h3>

-   Language Server version: 2021.6.3
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: strict

```python
from typing import Any

class AST: ...

class Foo(AST):

    value: str

class Bar(AST):

    value: AST

def handler(node: AST) -> Any:
    if isinstance(node, Foo):
        return node.value
    elif isinstance(node, Bar):
        if isinstance(node.value, Foo): # Unnecessary isinstance call; "str" is never an instance of "Foo" Pylance(reportUnnecessaryIsInstance)
            return node.value
```
this also.
```python
class Baz(AST):

    value: str

class Qux(AST):

    value: int

def handler(node: AST) -> Any:
    if isinstance(node, Baz):
        result = [node.value] # type: list[str] # Expression of type "list[str]" cannot be assigned to declared type "list[int]"
                                                #  TypeVar "_T@list" is invariant
                                                #    "str" is incompatible with "int"
        return result
    elif isinstance(node, Qux):
        result = [node.value] # type: list[int]
        return result
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202170-6-july-2021
